### PR TITLE
Remove old files before running tests

### LINKED ISSUE #361: Rerun of Advection Convergence test fails
The output directory of advection convergence tests needs to be cleaned before it is rerun, otherwise the test will fail. 

"advection_convergence test seems to not clean up advection-errors.dat file.  As a result running the test twice in a row causes it to fail."

### PR DESCRIPTION
**Pull request by JoshuaSBrown**
## PR Summary

Makes sure that tests are run in a clean folder. @nmsriram pointed out that the advection_convergence tests was failing if the folder was clean on a rerun. 

## PR Checklist

- [x] Code passes cpplint
- [ ] New features are documented.
- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md

### CODE REVIEW

**Review by Yurlungur — APPROVED** [review]

File: tst/regression/utils/test_case.py [hunk 1]
```diff
@@ -182,6 +183,9 @@ def MakeOutputFolder(self):
         if not os.path.isdir(self.parameters.output_path):
             os.makedirs(self.parameters.output_path)
 
+        files = glob.glob(self.parameters.output_path + "/*")
+        for f in files:
+            os.remove(f)
```
Comment (pgrete) [inline tst/regression/utils/test_case.py]: Could we potentially only clean the required files for specific tests, e.g., the `advection_errors.dat` for the advection test?
I always get goosebumps if automated pattern matching with a `*` is used to remove files in scripts/programs.

Comment (JoshuaSBrown) [reply tst/regression/utils/test_case.py]: Where would you put the logic? In test_case.py?

Comment (pgrete) [reply tst/regression/utils/test_case.py]: How about directly in the Prepare function of each test case like
```Python
if step == 1:
  magic_cleanup()
```

Comment (JoshuaSBrown) [reply tst/regression/utils/test_case.py]: @pgrete I can see having a clean output folder being a good default. How about I simply remove the named output folder, that will avoid the wildcard. We could always have a setting later that turns this default feature off.

Comment (JoshuaSBrown) [reply tst/regression/utils/test_case.py]: @pgrete

Comment (pgrete) [reply tst/regression/utils/test_case.py]: Sounds good. My concern was mostly about the wildcard itself ;)

**Review by AndrewGaspar — APPROVED** [review]

**Comment by JoshuaSBrown** [comment]
@pgrete can you approve this?

**Pull request merged**